internal/env/parser.go

```diff
@@ -112,6 +113,16 @@ func parseServerConfig(cfg *server.Config, altPrefix string, opts ...env.Options
 		if (cfg.Port == "" || cfg.Port == copyConfig.Port) && altCfg.Port != "" {
 			cfg.Port = altCfg.Port
 		}
+		if cfg.InstanceID == "" && altCfg.InstanceID != "" {
+			cfg.InstanceID = altCfg.InstanceID
+		}
+		if cfg.InstanceID == "" && altCfg.InstanceID == "" {
+			instanceID, err := uuid.New().ID()
```

Comment:
In previous line `altCfg.InstanceID` is assigned to `cfg.InstanceID`
So it sufficent to check only `cfg.InstanceID` is empty string 
```suggestion	
    if cfg.InstanceID == "" {
```